static/app/views/automations/components/dataConditionNodes.tsx

```diff
@@ -0,0 +1,143 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import NumberField from 'sentry/components/forms/fields/numberField';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import {t, tct} from 'sentry/locale';
+import {
+  type DataCondition,
+  DataConditionType,
+} from 'sentry/types/workflowEngine/dataConditions';
+
+export type Node = {
+  label: string;
+  configNode?: (
+    condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>,
+    condition_id: string,
+    onUpdate: (condition: Record<string, any>) => void
+  ) => React.ReactNode;
+};
+
+// TODO: split nodes into separate files
+export const dataConditionNodesMap: Partial<Record<DataConditionType, Node>> = {
+  [DataConditionType.FIRST_SEEN_EVENT]: {
+    label: t('A new issue is created'),
+  },
+  [DataConditionType.REGRESSION_EVENT]: {
+    label: t('A resolved issue becomes unresolved'),
+  },
+  [DataConditionType.REAPPEARED_EVENT]: {
+    label: t('An issue escalates'),
+  },
+  [DataConditionType.AGE_COMPARISON]: {
+    label: t('Compare the age of an issue'),
+    configNode: (
+      condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>,
+      condition_id: string,
+      onUpdate: (comparison: Record<string, any>) => void
+    ) => (
```

Comment:
I wonder if all of these args should actually be accessible from a `useDataConditionNodeContext()` hook so that the underlying components can remain modular?